### PR TITLE
Add support for Amber-lang

### LINKED ISSUE #12001: Add amber-lang support
Amber is
> The Programming Language compiled to Bash.
> -- https://amber-lang.com/

But currently in their alpha stage.

### PR DESCRIPTION
**Pull request by lens0021**
A screenshot of [install.ab](https://github.com/amber-lang/amber/blob/master/setup/install.ab):

image

The query files are from https://github.com/amber-lang/amber-zed/tree/b49e5a877872a7feb321723ca55bbd70f4e376b5/languages/amber.

Closes #12001

### CODE REVIEW

**Comment by lens0021** [comment]
@Ph0enixKM Hi, Could I use your code in this PR? If you are okay, I will mark this PR as ready for review.

**Comment by Ph0enixKM** [comment]
@lens0021 looking good... I'm just thinking how would the `highlights.scm` be synchronized with the current Amber version? Would you be updating it here?

**Comment by lens0021** [comment]
> @lens0021 looking good... I'm just thinking how would the `highlights.scm` be synchronized with the current Amber version later on in the future? Would you be updating it here?

I think it is common to manually updating the SHA in this repo. But I am not sure.

**Comment by lens0021** [comment]
(* so it is possible to sync the files using the same PR in manual way)

File: runtime/queries/amber/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+; Literals
+(boolean) @constant.builtin.boolean
+(number) @constant.numeric
+(null) @constant.numeric
+(string) @string
+(status) @keyword
+(command) @string
+(handler) @keyword
+(block) @delimeter
```
Comment (the-mikedavis) [inline runtime/queries/amber/highlights.scm:43]:
```suggestion
(block) @punctuation.delimiter
```

For our highlights queries we use these scopes: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

which I think are a little different than Zed's (but mostly the same). There's also a little typo there "delimeter=>delimiter". (Maybe that should be changed in the Zed queries.)

File: runtime/queries/amber/brackets.scm
Comment (the-mikedavis) [inline runtime/queries/amber/brackets.scm:1]: I think `brackets.scm` is specific to Zed. We don't currently have equivalent queries so this file can be removed

**Comment by the-mikedavis** [comment]
Updating `highlights.scm` is usually done manually when updating the revision in `languages.toml`. (For example the tree-sitter parser might expose new nodes to be highlighted and those would be added to the highlights when updating the revision.)

**Comment by lens0021** [comment]
rebased

File: runtime/queries/amber/brackets.scm
Comment (lens0021) [reply runtime/queries/amber/brackets.scm:1]: Removed.

File: runtime/queries/amber/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+(status) @keyword
+(command) @string
+(handler) @keyword
+(block) @punctuation.delimiter
+(variable_init) @keyword
+(variable_assignment) @delimiter
+(variable) @variable
+(escape_sequence) @constant.character.escape
+(type_name_symbol) @type
+(interpolation) @delimiter
```
Comment (the-mikedavis) [inline runtime/queries/amber/highlights.scm:49]:
```suggestion
(variable_assignment) @punctuation.delimiter
(variable) @variable
(escape_sequence) @constant.character.escape
(type_name_symbol) @type
(interpolation) @punctuation.delimiter
```

ah whoops there are two other `@delimiter`s to fix

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**